Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -608,3 +607,24 @@ void HexEditorWidget::drop_event(GUI::DropEvent& event)
         open_file(response.value().filename(), response.value().release_stream());
     }
 }
+
+DeprecatedString HexEditorWidget::make_display_string(DeprecatedString string)
+{
+    // FIXME: We probably want to replace all control characters here
+    StringBuilder builder;
+    for (auto c : string) {
+        if (c == '\t')
+            builder.append("␉"sv);
+        else if (c == '\n')
+            builder.append("␊"sv);
+        else if (c == '\v')
+            builder.append("␋"sv);
+        else if (c == '\f')
+            builder.append("␌"sv);
+        else if (c == '\r')
+            builder.append("␍"sv);
+        else
+            builder.append(c);
+    }
+    return builder.to_deprecated_string();
+}
```

Comment:
Please use the new non-deprecated `String` for new code, converting at the call site where needed

Reply:
it looks you might not need a builder here too (yet not 100% sure of that).

Reply:
@konradekk what would not using a `StringBuilder` look like here? It's not clear to me how it could reasonably be done otherwise

Reply:
@linusg OK I made `make_display_string()` return a `ErrorOr<String>`. It still accepts as input a `DeprecatedString` and I used `MUST()` at the call sites

Is this what you had in mind with your suggestion? Or were you also thinking all new code should not accept `DeprecatedString`?

Reply:
Changed `make_display_string()` to accept a `StringView` 👍